lib/experimental/block-editor-settings-mobile-allowed.php

```diff
@@ -0,0 +1,138 @@
+	 * Alternatively, nested structures can have nested allow-lists for their keys.
+	 *
+	 * @param array $initial_array Existing block editor settings.
+	 *
+	 * @param array $allow_list_array Structured allow-list.
+	 *
+	 * @return array New block editor settings.
+	 */
+	function filter_by_supported_block_editor_settings_mobile( $initial_array, $allow_list_array ) {
+		$result = array();
+
+		foreach ( $allow_list_array as $key => $value ) {
+			if ( ! array_key_exists( $key, $initial_array ) ) {
+				continue;
+			}
+
+			$initial_value = $initial_array[ $key ];
+
+			if ( array_key_exists( $key, $initial_array ) ) {
+				if ( is_array( $value ) && is_array( $initial_value ) ) {
+					$result[ $key ] = filter_by_supported_block_editor_settings_mobile( $initial_value, $value );
+				} else {
+					$result[ $key ] = $initial_value;
+				}
+			}
+		}
+
+		return $result;
+	}
+}
+
+if ( ! function_exists( 'gutenberg_get_allowed_block_editor_settings_mobile' ) ) {
+	/**
+	 * Keeps only supported settings for the mobile block editor.
+	 *
+	 * @param array $settings Existing block editor settings.
+	 *
+	 * @return array New block editor settings.
+	 */
+	function gutenberg_get_allowed_block_editor_settings_mobile( $settings ) {
+
+		if (
+			defined( 'REST_REQUEST' ) &&
+			REST_REQUEST &&
+			isset( $_GET['context'] ) &&
+			'mobile' === $_GET['context']
+		) {
+			return filter_by_supported_block_editor_settings_mobile(
+				$settings,
+				BLOCK_EDITOR_SETTINGS_MOBILE_ALLOW_LIST
+			);
+		} else {
+			return $settings;
+		}
+	}
+}
+
+if ( ! defined( 'BLOCK_EDITOR_SETTINGS_MOBILE_ALLOW_LIST' ) ) {
+	define(
+		'BLOCK_EDITOR_SETTINGS_MOBILE_ALLOW_LIST',
+		array(
+			'alignWide'                        => true,
+			'allowedBlockTypes'                => true,
+			'allowedMimeTypes'                 => true,
+			'defaultEditorStyles'              => true,
+			'blockCategories'                  => true,
+			'isRTL'                            => true,
+			'imageDefaultSize'                 => true,
+			'imageDimensions'                  => true,
+			'imageEditing'                     => true,
+			'imageSizes'                       => true,
+			'maxUploadFileSize'                => true,
+			'__unstableGalleryWithImageBlocks' => true,
+			'disableCustomColors'              => true,
+			'disableCustomFontSizes'           => true,
+			'disableCustomGradients'           => true,
+			'disableLayoutStyles'              => true,
+			'enableCustomLineHeight'           => true,
+			'enableCustomSpacing'              => true,
+			'enableCustomUnits'                => true,
+			'colors'                           => true,
+			'fontSizes'                        => true,
+			'__experimentalStyles'             => array(
+				'elements'   => true,
+				'spacing'    => true,
+				'blocks'     => true,
+				'color'      => true,
+				'typography' => true,
+			),
+			'__experimentalFeatures'           => array(
```

Comment:
Let's also add the `custom` attribute which is used for [custom variables or other settings](https://developer.wordpress.org/block-editor/how-to-guides/themes/theme-json/#css-custom-properties-presets-custom). The app uses it to get values for custom variables.